cmake/detect-intrinsics.cmake

```diff
@@ -364,6 +364,21 @@ macro(check_power8_intrinsics)
         }"
         HAVE_POWER8_INTRIN
     )
+    if (NOT HAVE_POWER8_INTRIN AND HAVE_LINUX_AUXVEC_H)
+        check_c_source_compiles(
+            "#include <sys/auxv.h>
+            #include <linux/auxvec.h>
+            int main() {
+                return (getauxval(AT_HWCAP2) & PPC_FEATURE2_ARCH_2_07);
+            }"
+            HAVE_POWER8_INTRIN2
+        )
+        if(HAVE_POWER8_INTRIN2)
+            set(POWER8_NEED_AUXVEC_H 1)
```

Comment:
Set to `ON` instead of `1` like we have many places elsewhere?

Reply:
Also why even set this `POWER9_NEED_AUXVEC_H` variable, it is not used further down in the file..

Reply:
It's used in root CMakeLists.txt to avoid adding same define twice if either POWER8 or POWER9 optimizations are disabled. I didn't use `ON` because this isn't a switch, it's internal variable.